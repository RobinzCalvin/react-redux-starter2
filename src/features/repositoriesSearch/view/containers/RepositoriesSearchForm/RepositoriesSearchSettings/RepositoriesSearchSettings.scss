@import '~shared/styles/mixins';

.repositories-search-settings {
  width: 12rem;

  @include breakpoint-up(sm) {
    width: 25rem;
  }

  &__row {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    margin-bottom: 0;

    @include breakpoint-up(sm) {
      flex-direction: row;
      margin-bottom: 1.5rem;
    }

    &:last-child {
      margin-bottom: 0;
    }
  }

  &__item {
    margin-bottom: 0.5rem;
    width: 100%;

    @include breakpoint-up(sm) {
      margin-bottom: 0;
      width: 11rem;
    }
  }
}
