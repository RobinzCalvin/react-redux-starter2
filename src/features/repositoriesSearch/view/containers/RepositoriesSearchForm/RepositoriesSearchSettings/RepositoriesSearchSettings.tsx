import * as React from 'react';
import block from 'bem-cn';

import { TextInputField, NumberInputField } from 'shared/view/form';

import { fieldNames } from '../constants';
import './RepositoriesSearchSettings.scss';

const b = block('repositories-search-settings');

function RepositoriesSearchSettings() {
  return (
    <div className={b()}>
      {renderFirstRowSettings()}
      {renderSecondRowSettings()}
    </div>
  );

  function renderFirstRowSettings() {
    return (
      <div className={b('row')}>
        <div className={b('item')}>
          <NumberInputField
            name={fieldNames.starsNumber}
            label="Min stars"
          />
        </div>
        <div className={b('item')}>
          <NumberInputField
            name={fieldNames.forksNumber}
            label="Min forks"
          />
        </div>
      </div>
    );
  }

  function renderSecondRowSettings() {
    return (
      <div className={b('row')}>
        <div className={b('item')}>
          <TextInputField
            name={fieldNames.language}
            label="Repositories language"
          />
        </div>
        <div className={b('item')}>
          <TextInputField name={fieldNames.owner} label="Owner" />
        </div>
      </div>
    );
  }
}

export default RepositoriesSearchSettings;
