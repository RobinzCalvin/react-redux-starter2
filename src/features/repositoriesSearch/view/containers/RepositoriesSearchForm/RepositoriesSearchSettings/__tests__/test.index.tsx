import React from 'react';
import { shallow } from 'enzyme';

import RepositoriesSearchSettings from '../RepositoriesSearchSettings';

describe('RepositoriesSearchSettings component', () => {
  const component = shallow(<RepositoriesSearchSettings />);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
