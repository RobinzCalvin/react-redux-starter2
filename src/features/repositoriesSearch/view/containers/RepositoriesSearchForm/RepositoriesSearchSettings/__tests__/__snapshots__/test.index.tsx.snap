// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesSearchSettings component should match snapshot 1`] = `
<div
  className="repositories-search-settings"
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <FieldWithComponent(NumberInputField)
        label="Min stars"
        name="starsNumber"
      />
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <FieldWithComponent(NumberInputField)
        label="Min forks"
        name="forksNumber"
      />
    </div>
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <FieldWithComponent(TextInputField)
        label="Repositories language"
        name="language"
      />
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <FieldWithComponent(TextInputField)
        label="Owner"
        name="owner"
      />
    </div>
  </div>
</div>
`;
